readme.md

```diff
@@ -0,0 +1,171 @@
+
+By using the SDK, the development of applications accessing Blue Button 2.0 API can be greatly simplified.
+
+Note, following the OAUTH2 best practices, OAUTH2 PKCE etension [(RFC 7636)](https://datatracker.ietf.org/doc/html/rfc7636) is always enabled.
+
+## Installation
+
+Using npm:
+
+```bash
+$ npm install cms-bluebutton-sdk
+```
+
+Using yarn:
+
+```bash
+$ yarn add cms-bluebutton-sdk
+```
+
+## Configuration
+
+the SDK needs to be properly configured to work, the parameters are:
+
+- the app's credentials - client id, client secret
+- the app's callback url
+- the version number of the API
+- the app's environment (web location where the app is registered)
+
+the configuration is in json format and stored in a local file, the default location
+is current working directory with file name: .bluebutton-config.json
+
+A sample configuration json:
+
+```
+{
+  "clientId": "foo",
+  "clientSecret": "bar",
+  "callbackUrl": "https://www.fake.com/",
+  "version": "1",
+  "environment": "PRODUCTION"
+}
+
+```
+
+| parameter    | value                   | Comments                                                                                                     |
+| ------------ | ----------------------- | ------------------------------------------------------------------------------------------------------------ |
+| clientId     | "foo"                   | oauth2 client id of the app                                                                                  |
+| clientSecret | "bar"                   | oauth2 client secret of the app                                                                              |
+| callbackUrl  | "https://www.fake.com/" | oauth2 callback url of the app                                                                               |
+| version      | 1 or 2                  | version number of BlueButton API (v1/v2)                                                                     |
+| environment  | PRODUCTION/SANDBOX      | the target URL:<br>PRODUCTION: https://api.bluebutton.cms.gov<br>SANDBOX: https://sandbox.bluebutton.cms.gov |
+
+## Usages
+
+### Sample Code
+
+Below are psuedo code snippets show SDK used with node express server.
+
+#### Main Usage: Obtain Access Grant and Access Data
+
+```
+
+import express, { Router, Request, Response } from 'express';
+
+const app = express();
+
+const bb = new BlueButton();
+const authData = bb.generateAuthData();
+
+// each associated with current logged in user in real app
+let authToken: AuthorizationToken;
+
+const router = Router();
+
+// auth flow: response with URL to redirect to Medicare.gov beneficiary login
+router.get('/authorize/authurl', bb.generateAuthorizeUrl);
+
+// auth flow: route oauth2 call back
+router.get('/bluebutton/callback', async (req: Request, res: Response) => {
+  authToken = await bb.getAuthorizationToken(authData, req.query.code, req.query.state, req.query.error);
+});
+
+// data flow
+
+// route get EOB requests
```

Comment:
I think we can probably remove these routes and instead just add the data pulls to the inside of the callback function. That I think is a more normal flow, generally I don't think apps build wrappers around our API endpoints.